sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -103,6 +97,9 @@ public class EventHubBufferedProducerClient : IAsyncDisposable
         /// <summary>A <see cref="CancellationTokenSource"/> instance to signal that any active publishing operations, including those in-flight, should terminate immediately.</summary>
         private CancellationTokenSource _activeSendOperationsCancellationSource;
 
+        /// <summary>A completion source that be awaited when publishing would like to pause and wait for an event to be enqueued.</summary>
+        private TaskCompletionSource<bool> _eventEnqueuedCompletionSource;
```

Comment:
It seems that the `bool` is not really used in the current implementation. Could this just be a `TaskCompletionSource` or are there reasons we care when it is set to `true` versus `false`?

Reply:
There's no non-generic `TaskCompletionSource` for `netstandard2.0` unfortunately.    _([ref](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcompletionsource.trysetresult?view=net-6.0&viewFallbackFrom=netstandard-2.0#system-threading-tasks-taskcompletionsource-trysetresult))_